orga/tests/integration/components/campaign/settings/view_test.js

```diff
@@ -207,15 +207,44 @@ module('Integration | Component | Campaign::Settings::View', function (hooks) {
   });
 
   module('on campaign url display', function () {
-    test('it should display the campaign url', async function (assert) {
-      // given
-      this.campaign = store.createRecord('campaign', { code: '1234' });
+    module('when current organization has not GAR as identity provider', function () {
+      test('it should display the campaign url', async function (assert) {
+        // given
+        this.campaign = store.createRecord('campaign', { code: '1234', identityProviderForCampaigns: '' });
```

Comment:
**Suggestion** : Ce test passe mais pas pour les bonnes raisons ^^.
Si on veut réellement représenter ce qu'il se passe dans le code, il faut représenter ça : this.currentUser.prescriber.hasCurrentOrganizationWithGARAsIdentityProvider;

Tu l'as très bien fait dans le test d'après, stubber le curentUser pour mettre un prescriber qui à l'info identityProviderForCampaigns. 

```
const organization = store.createRecord('organization', {
          name: 'Willow school',
          identityProviderForCampaigns: 'TUTU',
        });
        const userOrgaSettings = store.createRecord('userOrgaSetting', { organization });
        const prescriber = store.createRecord('prescriber', { userOrgaSettings });

        class CurrentUserStub extends Service {
          prescriber = prescriber;
        }
        this.owner.register('service:currentUser', CurrentUserStub);

        this.campaign = store.createRecord('campaign', { code: '1234' });
```

Reply:
Je corrige.
Mais du coup pourquoi ça passe alors ?

Reply:
J'imagine que ça passe parce que `identityProviderForCampaigns` est undefined du coup